client/src/redux/actions/userAction.js

```diff
@@ -10,12 +10,12 @@ export const updateUser = (id, formData) => async (dispatch) => {
   }
 }
 
-export const followUser = (id, data) => async (dispatch)=> {
+export const followUser = (id) => async (dispatch)=> {
+  await userApi.followUser(id)
```

Comment:
This is waiting to see if the API call is successful before continuing. If the API call fails then we don't really want to run the next line as it would put the frontend into an incorrect state (the database says the user is not followed, the frontend will say they are).